src/Features/CSharp/Portable/UseCompoundAssignment/CSharpUseCompoundAssignmentDiagnosticAnalyzer.cs

```diff
@@ -20,5 +20,9 @@ protected override SyntaxKind GetKind(int rawKind)
 
         protected override SyntaxKind GetAnalysisKind()
             => SyntaxKind.SimpleAssignmentExpression;
+
+        protected override bool IsSupported(SyntaxKind assignmentKind, ParseOptions options)
+            => assignmentKind != SyntaxKind.CoalesceExpression ||
+            ((CSharpParseOptions)options).LanguageVersion >= LanguageVersion.CSharp8;
```

Comment:
>LanguageVersion [](start = 61, length = 15)

Specifically this: the compiler already has this logic. We shouldn't be reimplementing it here.

Reply:
@Paxxi, note that I'm not asking this of you: I'm asking this of the compiler team :)

---
In reply to: [227126151](https://github.com/dotnet/roslyn/pull/30639#discussion_r227126151) [](ancestors = 227126151)

Reply:
> Specifically this: the compiler already has this logic. We shouldn't be reimplementing it here.

FWIW, this is exeedingly common in the IDE.  In general, there are no ways to ask te compiler "is this feature supported in this language version".  So, if we add support for some new feature, we just do the version check ourselves.

Reply:
Perhaps we should change this :)